pennylane_lightning/lightning_qubit.py

```diff
@@ -611,7 +610,24 @@ def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
         jac = jac.reshape(-1, len(trainable_params))
         jac_r = np.zeros((jac.shape[0], processed_data["all_params"]))
         jac_r[:, processed_data["record_tp_rows"]] = jac
-        return jac_r
+        return self._adjoint_jacobian_processing(jac_r) if qml.active_return() else jac_r
+
+    @staticmethod
+    def _adjoint_jacobian_processing(jac):
+        """
+        Post-process the Jacobian matrix returned by ``adjoint_jacobian`` for
+        the new return type system.
+        """
+        jac = np.squeeze(jac)
```

Comment:
Even though this is necessary now, do we expect much of an impact on runtime?

Reply:
From the docs:
```
The input array, but with all or a subset of the dimensions of length 1 removed. This is always a itself or a view into a.
```

It doesn't change the data itself, just the metadata that specifies its shape.